include/storage/flash_map.h

```diff
@@ -209,6 +209,18 @@ int flash_area_has_driver(const struct flash_area *fa);
  */
 struct device *flash_area_get_device(const struct flash_area *fa);
 
+#define FLASH_AREA_LABEL_EXISTS(label) \
+	DT_HAS_FIXED_PARTITION_LABEL(label)
+
+#define FLASH_AREA_ID(label) \
+	DT_FIXED_PARTITION_ID(DT_NODE_BY_FIXED_PARTITION_LABEL(label))
+
+#define FLASH_AREA_OFFSET(label) \
+	DT_REG_ADDR(DT_NODE_BY_FIXED_PARTITION_LABEL(label))
+
+#define FLASH_AREA_SIZE(label) \
+	DT_REG_SIZE(DT_NODE_BY_FIXED_PARTITION_LABEL(label))
+
```

Comment:
We should allow users to provide their own definition of these symbols. This could be the default definition included, but (like with `flash_map_default.c`) it should be possible to override this through a kconfig value.

Currently any dowstream with their own partition definitions is required to maintain a patch of this file.

Reply:
My initial thoughts for solving this were to in the future extend this macro to take a second parameter identifying the provider of partition information.  The default behavior would be as if that parameter were `fixed_partition` which would through magic pass to the current solution, but if it were say `partition_mgr` would route to a different macro or possibly ultimately a function.

The thought is that we do need label uniqueness within a given provider of flash region managers.  Attempting to enforce it across all region managers isn't reasonable IMO, and limiting applications to a single region manager (devicetree or other) is also not reasonable.

A way to resolve the conflict is to pair the label with a provider, using the existing devicetree provider as the default.

Reply:
This was consulted with me. @mbolivar This is the simple way we can inject our partitions info.

Reply:
OK, then there must be something like a kconfig option passed in as the second argument to avoid having to patch the file.

I'm fine with this as long as it allows us to not need patches in any of the upstream source files.